### PR TITLE
Quickstart: Build a decentralized app with Arbitrum (Solidity, Hardhat)

### PR DESCRIPTION
**Pull request by symbolpunk**
Currently E2E testing and polishing to "good enough" ~ ready for feedback!

 1. P0 nits
 2. Fact checking
 3. Strong opinions
 4. Weak opinions
 5. Broken links / quicklooks
 6. Ideas - how can we increase the probability that this becomes a "widely shared, world-class, canonical onboarding experience"?
 7. etc

staged preview [here](https://nitro-docs-git-boop-quickstart-offchain-labs.vercel.app/for-devs/quickstart-solidity-hardhat)

### CODE REVIEW

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+When our `VendingMachine` contract is deployed to Ethereum, it'll be hosted by Ethereum's decentralized network of nodes. We won't be able to modify the contract's code after it's deployed[^3].
+
+Let's convert our dumb contract into a smart contract.
+
+
+### Configure your project directory
+
+Create a `decentralized-cupcakes` directory for your project. Navigate to this directory from VS Code's integrated terminal and install [hardhat](https://hardhat.org/hardhat-runner/docs/getting-started#overview):
+
+<!-- todo: toggles for npm / yarn? -->
```
Comment (DZGoldman) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:118]: Given the instruction granularity earlier (i.e., installing IDE) might want to include yarn installation / directory set up, e.g.

- Install yarn https://classic.yarnpkg.com/lang/en/docs/install/#mac-stable
- make a new directory `mkdir my_cupcake_app`
- setup new yarn project `yarn init`

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+  await vendingMachine.deployed();
+
+  console.log(
+    `Cupcake vending machine deployed to ${vendingMachine.address}`
+  );
+}
+
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
```
Comment (DZGoldman) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:159]: I believe this needs the additional line `process.exit()`

Or alternatively, just replace both w/ the the one-line

`process.exit(1)`

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+pragma solidity ^0.8.9;
+
+// Rule 2: The vending machine's rules can't be changed by anyone.
+contract VendingMachine {
+    // state variables = internal memory of the vending machine
+    mapping(address => uint) private _cupcakeBalances;
+    mapping(address => uint) private _cupcakeDistributionTimes;
+
+    function giveCupcakeTo(address userAddress) public returns (bool) {
+        if (_cupcakeBalances[userAddress] == 0) {
```
[the remark this answers is not in the document]
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:201]: ty ~ noted intent inline ~ I'm split between "1:1 mapping between js:solidity for the learning", and "succinct-solidity" - will feel it out as I approach content completeness 🙏

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+...etc
+```
+
+<!-- todo: psst, don't ever share your mainnet wallet private keys or even copy them into your clipboard -->
+
+Note that in the context of this quickstart, "account" refers to a public wallet address and its associated private key.
+
+#### Configure Metamask
+
+Next, open Metamask and create or import a wallet by following the displayed instructions. By default, Metamask will connect to Ethereum mainnet. To connect to our local "testnet", enable test networks through Metamask's settings. Then select the `Localhost 8545` network:
```
[the remark this answers is not in the document]
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:255]: IIRC a fresh installation of metamask displayed it by default for me as of last week... will leave this open until I can confirm 🙏

[the remark this answers is not in the document]
Comment (DZGoldman) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:255]: ah okay cool, i could just be wrong

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+
+// Rule 2: The vending machine's rules can't be changed by anyone.
+contract VendingMachine {
+    // state variables = internal memory of the vending machine
+    mapping(address => uint) private _cupcakeBalances;
+    mapping(address => uint) private _cupcakeDistributionTimes;
+
+    function giveCupcakeTo(address userAddress) public returns (bool) {
+        if (_cupcakeBalances[userAddress] == 0) {
+            _cupcakeDistributionTimes[userAddress] = block.timestamp; // todo - decide whether or not there should be 1:1-ish mapping between js and solidity "contracts" for the sake of pattern recognition / learning; if so, mirror; if not, delete
```
Comment (DZGoldman) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:202]: actually looking again: as-is it's a bug, not just a noop  — the initial attempt to claim will always fail, (block.timestamp + 5 seconds <= block.timestamp)

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+        }
+
+        // Rule 1: The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+        uint fiveSeconds = 5 * 1 seconds;
+        bool userCanReceiveCupcake = _cupcakeDistributionTimes[userAddress] + fiveSeconds <= block.timestamp;
+        if (userCanReceiveCupcake) {
```
Comment (DZGoldman) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:208]: if/else revert can be replace with a single require statement

Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:208]: tyty... original (possibly suboptimal) intent is to illustrate a 1:1 mapping between js:sol contracts so devs see how similar the syntax is... thinking maybe we can literally drop these comments into the code to get the best of both worlds with minimal copy / duplication.... will tinker with this 🙏

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+We're going to build a digital cupcake vending machine using Solidity smart contracts[^1]. Our vending machine will follow two rules:
+
+1. The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+2. The vending machine's rules can't be changed by anyone.
+
+Here's our vending machine implemented with Javascript:
+
+import { VendingMachine } from '@site/src/components/VendingMachine/VendingMachine'
+
+<VendingMachine id='dumb-cupcakes' type='web2' />
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:41]: I would add an example in the placeholder, like "Enter identity (e.g. your name)" or similar. I got confused for a second about what "identity" I was supposed to enter.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 8]
```diff
@@ -0,0 +1,368 @@
...
+      url: "https://arbnova.com/rpc",
+      //accounts: [process.env.PRIVATE_KEY]
+    }
+  }
+};
+```
+
+Run `npx hardhat compile` to compile the default `contracts`. You may be prompted to install additional dependencies - follow those instructions until this command runs successfully[^6]. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/` directory. This directory contains the compiled smart contract[^7].
+
+Open `scripts/deploy.js` and replace its contents with the following:
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:167]: Is it weird if I find it confusing to edit deploy.js before VendingMachine.sol ? I'm thinking about following a logic similar to "set config => write contract => deploy it". Disregard if it looks clear enough as it is.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+ - Oracles
+ - One vs Nova
+ - https://yos.io/2019/11/10/smart-contract-development-best-practices/
+// https://github.com/smartcontractkit/full-blockchain-solidity-course-js/discussions/4733
+
+
+[^1]: Inspired by [Ethereum.org's Introduction to Smart Contracts](https://ethereum.org/en/developers/docs/smart-contracts/), which was inspired by [Nick Szabo's From vending machines to smart contracts](http://unenumerated.blogspot.com/2006/12/from-vending-machines-to-smart.html).
+[^2]: Although application front-ends are usually hosted by centralized services, smart contracts allow the underlying logic and data to be partially or fully decentralized. These smart contracts are hosted by Ethereum's public, decentralized network of nodes. This means that instead of asking users to trust centralized service providers, web3 apps ask users to trust Ethereum's decentralized network of nodes and the open source client software that they use to process and validate transactions.
+[^3]: todo
+[^4]: When our `VendingMachine` contract is deployed to Ethereum, it'll be hosted by Ethereum's decentralized network of nodes. We won't be able to modify the contract's code after it's deployed.
```
[the remark this answers is not in the document]
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:363]: (copying here just to make sure I don't resolve this until we align)

> how much of a lift would it be to read the href from the footnote superscript elements, and then display a popover that pulls html from the corresponding footnote's li (based on the id)?

> I feel that friction of being pulled down and then having to scroll back up... sucks. maybe we could do the quicklook thing post-launch if it's not ez

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 8 above)
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:167]: It's not weird it's a wonderful nudge, updating ^^

ty 🙏

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 10]
```diff
@@ -0,0 +1,368 @@
...
+1. The vending machine will distribute a cupcake to anyone who hasn't recently received one.
+2. The vending machine's rules can't be changed by anyone.
+
+Here's our vending machine implemented with Javascript:
+
+import { VendingMachine } from '@site/src/components/VendingMachine/VendingMachine'
+
+<VendingMachine id='dumb-cupcakes' type='web2' />
+
+Note that although this vending machine follows Rule 1, it doesn't follow Rule 2 (yet)[^2].
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:43]: Should we explicitly mention why it doesn't follow rule 2? (It is described below, but at this point it might make the user feel a bit lost because it might not understand how/why the rules can be changed).

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 11]
```diff
@@ -0,0 +1,368 @@
...
+
+### Configure your project directory
+
+Create a `decentralized-cupcakes` directory for your project and install [hardhat](https://hardhat.org/hardhat-runner/docs/getting-started#overview)[^6] using VS Code's integrated terminal:
+
+```
+mkdir decentralized-cupcakes
+cd decentralized-cupcakes
+yarn init -y
+yarn add hardhat @nomicfoundation/hardhat-toolbox
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:120]: Minor detail: Add "-D" at the end of the "yarn add" command, to install hardhat as a devDependency, like they suggest in their docs.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 12]
```diff
@@ -0,0 +1,368 @@
...
+```
+
+This installs two packages: `hardhat` lets us test and deploy our smart contracts, and `hardhat-toolbox` is a bundle of popular Hardhat plugins that we'll use later.
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:123]:
```suggestion
This installs two packages: `hardhat` lets us write, test and deploy our smart contracts, and `hardhat-toolbox` is a bundle of popular Hardhat plugins that we'll use later.
```

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 13]
```diff
@@ -0,0 +1,368 @@
...
+
+Next, run `npx hardhat` to configure Hardhat. Select `Create a JavaScript project` when prompted. Make sure you specify your `decentralized-cupcakes` directory as the project root when asked.
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:125]: I would stick with one package manager: either yarn or npm for the whole tutorial. Seeing that you only mention yarn when installing the package, it might be easier to just replace that with npm.
The alternative is to replace "npx" with "yarn" here and in the next occurrences.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 14]
```diff
@@ -0,0 +1,368 @@
...
+
+Note that in the context of this quickstart, "account" refers to a public wallet address and its associated private key.
+
+#### Configure Metamask
+
+Next, open Metamask and create or import a wallet by following the displayed instructions. By default, Metamask will connect to Ethereum mainnet. To connect to our local "testnet", enable test networks through Metamask's settings. Then select the `Localhost 8545` network:
+
+![Connect Metamask to Localhost 8545](assets/quickstart-localhost-metamask.png)
+
+Your mainnet wallet won't have a balance on your local testnet's node, but we can import one of the test accounts into Metamask to gain access to 10,000 fake ETH. Copy the private key of one of the test accounts (**excluding** the `0x` prefix, so eg `ac0..f80`) and import it into Metamask:
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:259]: Access to the private key is a bit hidden in Metamask, so maybe we should clarify this further => 3-dots / Account details / Export private key.
Also, private key does not have the "0x" prefix in Metamask.

Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:259]: This is instructing the developer to import the test account into metamask, from the local node's console output, and then removing the `0x` prefix before importing (procedure verified as of last week, Matamask vLatest). Can you clarify your feedback?

Is this not clear, when reviewing the content in its [rendered preview](https://nitro-docs-git-boop-quickstart-offchain-labs.vercel.app/for-devs/quickstart-solidity-hardhat) form? If not, I'll iterate until it is :).

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 13 above)
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:125]: For now, the most valuable form of feedback that you can provide is feedback related to the broad strokes of the procedure, and the code - I may end up implementing toggles so users can switch between consistent `npm` vs `yarn` instructions, but that's TBD, and in the meantime I'm recording my procedure :). 

I'll flip this from `Draft` to `Open` when I'm ready for micro feedback. Thank you!

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 14 above)
Comment (TucksonDev) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:259]: Oh, you're right, sorry about that. I don't know why I thought it was doing the operation the other way around (Metamask => hardhat). It is well explained as it is.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 8 above)
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:173]: Ended up on this order of operations:

1. Review JS
2. Configure project directory (all hardhat concerns here - installation and config)
3. Compile defaults
4. Compile VendingMachine
5. Deploy locally
  1. Run hardhat
  2. Configure metamask (all metamask concerns here)
  3. Deploy & test
6. Update config, deploy to testnet
7. Update config, deploy to mainnet

File: website/src/css/custom.less [hunk 15]
```diff
@@ -203,7 +193,158 @@ article {
...
+    font-size: 12px;
+    margin-top: 20px;
+    margin-bottom: 30px;
+
+    td {
+        padding: 10px;
+    }
+}
+
+[class*="quickstart-solidity-hardhat"] {
```
Comment (symbolpunk) [inline website/src/css/custom.less:216]: This ensures that the custom styling only applies to the quickstart. If we want to see these styles applied more consistently throughout the docs, we could easily do that (ideally via another PR).

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 16]
```diff
@@ -0,0 +1,493 @@
...
+
+Select `Arbitrum Goerli` from Metamask's dropdown, paste your contract address into the `VendingMachine` below, and click `Get cupcake!`. You should be prompted to sign a transaction that gives you a cupcake.
+
+<VendingMachine id='smart-goerli-cupcakes' type='web3-arb-goerli' />
+
+
+### Deploy the smart contract to Arbitrum One Mainnet
+
+
+First, update the `hardhat.config.js` file to specify the private key of the **one-time-use deployment account** that you'll use to deploy your smart contract (and pay the transaction fee):
```
Comment (symbolpunk) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:367]: Not confident about this guidance ~

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 17]
```diff
@@ -0,0 +1,491 @@
...
+### Ethereum and Arbitrum in a nutshell
+
+ - Ethereum is a decentralized network of <a href='https://docs.prylabs.network/docs/concepts/nodes-networks'>nodes</a> that use Ethereum's client software (like Offchain's <a href='https://docs.prylabs.network/docs/getting-started'>Prysm</a>) to maintain a public <a data-quicklook-from='blockchain'>blockchain</a> data structure.
+ - The data within Ethereum's blockchain data structure changes one transaction at a time.
+ - <a data-quicklook-from='smart-contract'>Smart contracts</a> are small programs that execute transactions according to predefined rules. Ethereum's nodes host and execute smart contracts.
+ - You can use smart contracts to build decentralized apps (dApps) that use Ethereum's network to process transactions and store data.
+ - DApps let users carry their data and identity between applications without having to trust centralized service providers.
+ - People who run Ethereum nodes[^3] can receive rewards for processing and validating transactions on behalf of users and dApps.
+ - These transactions can be expensive when the network is under heavy load. **Layer 2** (L2) scaling solutions like Arbitrum solve this problem.
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:58]: Should we probably specify that Arbitrum is an _optimistic rollup_ among different types of the L2 scaling solutions?

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 18]
```diff
@@ -0,0 +1,491 @@
...
+Note that we're adding a private key to a config file. This is **not** a best practice. In a real application, you should use a [signer](https://docs.ethers.io/v5/api/signer/) to sign transactions instead of handling private keys directly.
+
+:::
+
+Next, let's deposit some $AGOR into the wallet corresponding to the private key we added to `hardhat.config.js`. There are a few ways to acquire $AGOR:
+
+1. Use an L2 Goerli ETH faucet.
+2. Use an L1 Goerli ETH faucet and bridge your L1 Goerli ETH into Arbitrum L2 using [the Arbitrum bridge](https://bridge.arbitrum.io/).
+   - [goerlifaucet.com](https://goerlifaucet.com/) is a popular L1 Goerli ETH faucet.
+3. Ask someone nicely on [Discord](https://discord.gg/ZpZuw7p)[^6].
```
Comment (mahsamoosavi) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:340]: I'm wondering if we want to push people to ask us for faucets on Discord or not :?

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 19]
```diff
@@ -0,0 +1,491 @@
...
+
+### Prerequisites
+
+- **VS Code**: The IDE we'll use to build our vending machine. See [code.visualstudio.com](https://code.visualstudio.com/) to install.
+- **Metamask**: The <a data-quicklook-from='ethereum-wallet'>wallet</a> we'll use to interact with our vending machine. See [metamask.io](https://metamask.io/) to install.
+- **Yarn**: The package manager we'll use to install our dependencies. See [yarnpkg.com](https://yarnpkg.com/) to install.
+
+We'll install the rest of our dependencies as we go.
+
+### Ethereum and Arbitrum in a nutshell
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:50]: I'm wondering if we could split this section in 2 (or 2 subsections): one for Ethereum and one for Arbitrum. If a dev is coming from web2, it might be confusing to talk about Ethereum this much in this section, and just mention Arbitrum in the last 2 points.
What do you think?

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 20]
```diff
@@ -0,0 +1,491 @@
...
+    }
+
+    // Getter function for the cupcake balance of a user
+    function getCupcakeBalanceFor(address userAddress) public view returns (uint) {
+        return _cupcakeBalances[userAddress];
+    }
+}
+```
+
+Note that this smart contract is written in Solidity, a language that compiles to EVM bytecode. This means that it can be deployed to any Ethereum-compatible blockchain, including Ethereum mainnet, <a data-quicklook-from='arbitrum-one'>Arbitrum One</a>, and <a data-quicklook-from='arbitrum-nova'>Arbitrum Nova</a>.
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:217]: Should we add some reference for "EVM bytecode" here?
Maybe a glossary term or a link that explains what it is.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 18 above)
Comment (TucksonDev) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:340]: Agree. Also, if there isn't any L2 Goerli faucet I wouldn't mention it here. It might also lead users to ask about it on Discord.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 21]
```diff
@@ -0,0 +1,491 @@
...
+// ...
+const GOERLI_TESTNET_PRIVATE_KEY = ""; // <- this should **not** begin with "0x"
+// ...
+accounts: [GOERLI_TESTNET_PRIVATE_KEY] // <- uncomment this line
+// ...
+```
+
+:::caution
+
+Note that we're adding a private key to a config file. This is **not** a best practice. In a real application, you should use a [signer](https://docs.ethers.io/v5/api/signer/) to sign transactions instead of handling private keys directly.
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:331]: I'm not sure this is technically correct. You are already using a signer in the code (to deploy the contract, iirc). But, it's true that devs should use other means to store their keys. We can mention "environment variables" as an example. 
Actually, in a future iteration we could use this method in this quickstart, instead of describing it as an alternative here.

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 22]
```diff
@@ -0,0 +1,491 @@
...
+| **Money**             | Devs and users pay centralized service providers for server access using fiat currency. | <- same, but only for the presentation-layer concerns (code that supports frontend UI/UX). | <- same, but devs and users pay **testnet $ETH** to testnet validators.                | <- same, but instead of testnet $ETH, they use **mainnet $ETH**.                          |
+
+
+Next, we'll deploy our smart contract to a network of real nodes: Arbitrum's Goerli testnet.
+
+### Deploy the smart contract to the Arbitrum Goerli testnet
+
+We were able to deploy to a local testnet for free because we were using [Hardhat's built-in Ethereum network emulator](https://hardhat.org/hardhat-network/docs/overview#hardhat-network). Because Arbitrum's Goerli testnet is powered by a real network of real nodes, we'll need to pay a small transaction fee to deploy our smart contract. This fee will be paid with the Arbitrum Goerli testnet's token, $AGOR.
+
+:::info $AGOR ISN'T REAL
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:313]: I'm not sure this claim is accurate. We could say $AGOR is real and is a "token", as much as "ETH" is. The difference is that it does not hold value.
(And even that is not technically correct now, but I think it's safe to say)

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md [hunk 23]
```diff
@@ -0,0 +1,491 @@
...
+```
+
+This tells hardhat to deploy the compiled smart contract through the RPC endpoint corresponding to `arbitrumGoerli` in `hardhat.config.js`. You should see the following output:
+
+
+```bash
+Cupcake vending machine deployed to 0xff825139321bd8fB8b720BfFC5b9EfDB7d6e9AB3
+```
+
+Congratulations! You've just deployed **business logic and data** to Arbitrum Goerli. This logic and data will find its way to Ethereum's L1 Goerli testnet, where it'll be hashed and stored in a Merkle tree and mirrored across all nodes in the Goerli network.
```
Comment (TucksonDev) [inline arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:355]: `find its way to Ethereum's L1 Goerli testnet`
This might be confusing if the reader doesn't know how Arbitrum works. Maybe we can add a note with a link to how Arbitrum posts the information on L1?

File: arbitrum-docs/for-devs/quickstart-solidity-hardhat.md (hunk 20 above)
Comment (symbolpunk) [reply arbitrum-docs/for-devs/quickstart-solidity-hardhat.md:217]: TODO (eventually): A new concept doc that ELI5 -> ELI30s this thing in the context of Arbitrum (and Stylus)

(I'll find something else to link to in the meantime)

**Review by TucksonDev — APPROVED** [review]
